runtime/wasm/src/host_exports.rs

```diff
@@ -775,23 +904,85 @@ impl HostExports {
     ) -> Option<serde_json::Map<String, serde_json::Value>> {
         block_on03(self.three_box_adapter.profile(address)).ok()
     }
+
+    pub(crate) fn json_from_bytes(
+        &self,
+        bytes: &Vec<u8>,
+        gas: &GasCounter,
+    ) -> Result<serde_json::Value, DeterministicHostError> {
+        gas.consume_host_fn(gas::DEFAULT_GAS_OP.with_args(gas::complexity::Size, &bytes))?;
```

Comment:
I expect this should have a reasonably expensive OP, since parsing JSON is slow and hard. Comparing to h160 parsing (below) - the latter is practically free.

Reply:
True that the weights are not comparable, but I think the default throughput assumption of 10MB/s is pessimistic enough to be a lower bound to the performance of json parsing.